csi/controller_server.go

```diff
@@ -635,6 +645,29 @@ func (cs *ControllerServer) createCSISnapshotTypeLonghornBackup(req *csi.CreateS
 		return nil, status.Error(codes.InvalidArgument, "Snapshot name must be provided")
 	}
 
+	volume, err := cs.apiClient.Volume.ById(csiVolumeName)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to get volume %v", csiVolumeName)
+	}
+	if volume.State == string(longhorn.VolumeStateDetached) {
+		log := logrus.WithFields(
+			logrus.Fields{
+				"CSISnapshot": "auto-attach",
```

Comment:
NIT: This k-v pair looks weird. I don't think we need it.

Reply:
Removed.